src/Players.sol

```diff
@@ -6,26 +6,79 @@ import { ERC721 } from "../lib/openzeppelin-contracts/contracts/token/ERC721/ERC
 import { ERC721Enumerable } from "../lib/openzeppelin-contracts/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
 import { Base64 } from "../lib/openzeppelin-contracts/contracts/utils/Base64.sol";
 import { Strings } from "../lib/openzeppelin-contracts/contracts/utils/Strings.sol";
+import { ITerminus } from "../lib/web3/contracts/interfaces/ITerminus.sol";
 
 contract BeerLeagueBallers is ERC721Enumerable {
-    string[24] public ProfileImages = [
-        "https://badges.moonstream.to/blb/p0.png",
-        "https://badges.moonstream.to/blb/p1.png",
-        "https://badges.moonstream.to/blb/p2.png",
-        "https://badges.moonstream.to/blb/p3.png",
-        "https://badges.moonstream.to/blb/p4.png",
-        "https://badges.moonstream.to/blb/p5.png",
-        "https://badges.moonstream.to/blb/p6.png",
-        "https://badges.moonstream.to/blb/p7.png"
-    ];
+    mapping(uint256 => string) public ProfileImages;
+    uint256 public NumProfileImages;
 
     mapping(uint256 => string) public Name;
     mapping(uint256 => uint256) public ImageIndex;
 
-    constructor() ERC721("Beer League Ballers", "BLB") { }
+    address adminTerminus;
+    uint256 adminPoolID;
+
+    constructor(address _adminTerminus, uint256 _adminPoolID) ERC721("Beer League Ballers", "BLB") {
+        adminTerminus = _adminTerminus;
+        adminPoolID = _adminPoolID;
+
+        ProfileImages[0] = "https://s3.amazonaws.com/static.fullcount.xyz/Beer_League_Ballers/p0.png";
+        ProfileImages[1] = "https://s3.amazonaws.com/static.fullcount.xyz/Beer_League_Ballers/p1.png";
+        ProfileImages[2] = "https://s3.amazonaws.com/static.fullcount.xyz/Beer_League_Ballers/p2.png";
+        ProfileImages[3] = "https://s3.amazonaws.com/static.fullcount.xyz/Beer_League_Ballers/p3.png";
+        ProfileImages[4] = "https://s3.amazonaws.com/static.fullcount.xyz/Beer_League_Ballers/p4.png";
+        ProfileImages[5] = "https://s3.amazonaws.com/static.fullcount.xyz/Beer_League_Ballers/p5.png";
+        ProfileImages[6] = "https://s3.amazonaws.com/static.fullcount.xyz/Beer_League_Ballers/p6.png";
+        ProfileImages[7] = "https://s3.amazonaws.com/static.fullcount.xyz/Beer_League_Ballers/p7.png";
+        NumProfileImages = 8;
+    }
+
+    function _enforceIsAdmin() internal view {
+        ITerminus terminus = ITerminus(address(adminTerminus));
+        require(terminus.balanceOf(msg.sender, adminPoolID) > 0, "BeerLeagueBallers._enforceIsAdmin: not admin");
+    }
+
+    function addProfileImage(string memory newImage) public {
```

Comment:
We're probably going to be adding multiple images at once, this seems like it should be a batch method.

Reply:
We decided not to do this.